### DISCUSSION TITLE
AuthorizationStrategy interface: Option for checking context against a cloud service id directly

### DISCUSSION THREAD

**Discussion by lebogg** [clouditor/clouditor#1078] [Ideas]
Current implementation:
https://github.com/clouditor/clouditor/blob/8fdd407a0a8433a1090965408eda21c9bcd21f48/service/authorization.go#L55-L58

Couldn't we add the option to directly check against a targeted cloud service id? Sth. like:
```
CheckAccessWithCloudServiceID(ctx context.Context, typ RequestType, cloudServiceID string) bool
```
This would be a shortcut to the usage of `AllowedCloudServices`.

An alternative would be to set the last parameter of`CheckAccess` to an option (`WithRequestOption` and `WithServiceIDOPtion`).  I was just thinking, maybe it is to much ?

@oxisto

**Comment by oxisto** [comment]
You can already do that. maybe `api.CloudServiceRequest` is a misleading interface name now. It basically just contains of `GetCloudServiceId()`. As long as your service implements that function, you can check against it.
